test/service/test_repository_proxy.py

```diff
@@ -81,6 +108,84 @@ def assertUrlEqual(self, a: furl, b: furl):
         self.assertEqual(a.path, b.path)
         self.assertEqual(sorted(a.args.allitems()), sorted(b.args.allitems()))
 
+
+class TestTDRRepositoryProxy(RepositoryPluginTestCase):
+    catalog = 'test-tdr'
+    catalog_config = f'hca:{catalog}:metadata/hca:repository/tdr'
+
+    @mock.patch.dict(os.environ,
+                     AZUL_TDR_SERVICE_URL=mock_tdr_service_url,
+                     AZUL_TDR_SOURCES=mock_tdr_sources)
+    @mock.patch.object(type(TerraClient()),
+                       'oauthed_http',
+                       AuthorizedHttp(MagicMock(),
+                                      urllib3.PoolManager(ca_certs=certifi.where())))
+    def test_repository_files_proxy(self):
+        client = http_client()
+
+        file_uuid = '701c9a63-23da-4978-946b-7576b6ad088a'
+        file_version = '2018-09-12T121154.054628Z'
+        organic_file_name = 'foo.txt'
+        drs_path_id = 'v1_c99baa6f-24ce-4837-8c4a-47ca4ec9d292_b967ecc9-98b2-43c6-8bac-28c0a4fa7812'
+        file_doc = {
+            'name': organic_file_name,
+            'version': file_version,
+            'drs_path': drs_path_id,
+        }
+        for fetch in True, False:
+            with self.subTest(fetch=fetch):
+                with mock.patch.object(IndexQueryService,
+                                       'get_data_file',
+                                       return_value=file_doc):
+                    azul_url = furl(
+                        url=self.base_url,
+                        path='/fetch/repository/files' if fetch else '/repository/files',
+                        args={
+                            'catalog': self.catalog,
+                            'version': file_version,
+                        }).add(path=file_uuid)
+
+                    gs_bucket_name = 'gringotts-wizzard-bank'
```

Comment:
Awh, now I've expose that I am not a true fan :'(

Reply:
I only noticed because "Wizard" was misspelled and I had to double check the misspelling wasn't also present in the books. I haven't read/watched Harry Potter in like 10 years.